src/core-services/tags/queries/tag.js

```diff
@@ -17,16 +17,8 @@ export default async function tag(context, input) {
   const { Tags } = collections;
   const { slugOrId, shopId, shouldIncludeInvisible = false } = input;
 
-  // Check to make sure user has `read` permissions for this tag
-  // TODO(auth-pod): revisit this check once legacyRoles are removed
-  // await context.validatePermissions("reaction:legacy:tags:${slugOrId}", "read", {
-  //   shopId,
-  //   legacyRoles: ["admin", "owner", "tags", "any"]
-  // });
-
   // Check to see if user has `read` permissions for hidden / deleted tags
-  // TODO(pod-auth): revisit using `inactive` in resource, and revisit the word `inactive`
-  const hasInactivePermissions = await context.userHasPermission(`reaction:legacy:tags-inactive:${slugOrId}`, "read", {
+  const hasInactivePermissions = await context.userHasPermission(`reaction:legacy:tags:${slugOrId}`, "read:inactive", {
```

Comment:
With having slugOrId in the resource, does that mean we want to create a resource for every slug, too, and re-create when the slug changes? Or would it be better to rewrite this to do the db lookup by slug first, and then grab the ID and only ever check permission on the ID?

Reply:
Good catch. I think you're right, that we would need to create a resource for every slug too, and update it, which I don't think is the best idea. The initial DB lookup is the best way to do this.

Reply:
https://github.com/reactioncommerce/reaction/pull/6031/commits/69540767c6177e05f615343918a4b3111f67a9bc